contracts/liquid-locking/src/lib.rs

```diff
@@ -0,0 +1,149 @@
+#![no_std]
+
+multiversx_sc::imports!();
+
+#[multiversx_sc::contract]
+pub trait LiquidLocking {
+    #[init]
+    fn init(&self, unbond_period: u64) {
+        self.unbond_period().set_if_empty(unbond_period);
+    }
+
+    #[only_owner]
+    #[endpoint]
+    fn set_unbond_period(&self, unbond_period: u64) {
+        self.unbond_period().set(unbond_period);
+    }
+
+    #[only_owner]
+    #[endpoint]
+    fn whitelist_token(&self, token: TokenIdentifier) {
+        require!(token.is_valid_esdt_identifier(), "invalid token provided");
+        let _ = self.token_whitelist().insert(token);
+    }
+
+    #[only_owner]
+    #[endpoint]
+    fn blacklist_token(&self, token: &TokenIdentifier) {
+        let _ = self.token_whitelist().swap_remove(token);
+    }
+
+    #[payable("*")]
+    #[endpoint]
+    fn stake(&self) {
```

Comment:
If we renamed the file to locking, let's also rename the endpoints and functions here